test/Microsoft.ComponentDetection.Detectors.Tests/Utilities/SimplePypiClientTests.cs

```diff
@@ -0,0 +1,366 @@
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.ComponentDetection.Common;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Detectors.Pip;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Moq.Protected;
+using Newtonsoft.Json;
+
+[TestClass]
+public class SimplePyPiClientTests
+{
+    private readonly SimplePyPiClient simplePypiClient;
+
+    public SimplePyPiClientTests() => this.simplePypiClient = new SimplePyPiClient(
+        new EnvironmentVariableService(),
+        new Mock<ILogger<SimplePyPiClient>>().Object);
+
+    private Mock<HttpMessageHandler> MockHttpMessageHandler(string content, HttpStatusCode statusCode)
+    {
+        var handlerMock = new Mock<HttpMessageHandler>();
+        handlerMock.Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),
+                ItExpr.IsAny<CancellationToken>())
+            .ReturnsAsync(new HttpResponseMessage()
+            {
+                StatusCode = statusCode,
+                Content = new StringContent(content),
+            });
+
+        return handlerMock;
+    }
+
+    [TestMethod]
+    public async Task GetSimplePypiProject_DuplicateEntries_CallsGetAsync_OnceAsync()
+    {
+        var pythonSpecs = new PipDependencySpecification { DependencySpecifiers = new List<string> { "==1.0.0" }, Name = "boto3" };
+        var pythonProject = this.SampleValidApiJsonResponse("boto3", "0.0.1");
+        var mockHandler = this.MockHttpMessageHandler(pythonProject, HttpStatusCode.OK);
+        SimplePyPiClient.HttpClient = new HttpClient(mockHandler.Object);
+
+        Func<Task> action = async () => await this.simplePypiClient.GetSimplePypiProjectAsync(pythonSpecs);
+
+        await action.Should().NotThrowAsync();
+        await action.Should().NotThrowAsync();
+
+        // Verify the API call was performed only once
+        mockHandler.Protected().Verify(
+            "SendAsync",
+            Times.Once(),
+            ItExpr.IsAny<HttpRequestMessage>(),
+            ItExpr.IsAny<CancellationToken>());
+    }
+
+    [TestMethod]
+    public async Task GetSimplePypiProject_DifferentEntries_CallsGetAsync_TwiceAsync()
+    {
+        var pythonSpecs = new PipDependencySpecification { DependencySpecifiers = new List<string> { "==1.0.0" } };
+        var pythonProject = new SimplePypiProject()
+        {
+            Files = new List<SimplePypiProjectRelease> { new SimplePypiProjectRelease() },
+        };
+
+        var mockHandler = this.MockHttpMessageHandler(JsonConvert.SerializeObject(pythonProject), HttpStatusCode.OK);
+        SimplePyPiClient.HttpClient = new HttpClient(mockHandler.Object);
+
+        Func<Task> action = async () =>
+        {
+            pythonSpecs.Name = Guid.NewGuid().ToString();
```

Comment:
Extract this line above. It doesn't need to be in the function.

Reply:
I think it actually should be in the function since the test is to make sure the request isnt pulling a cached value, and having the spec name in the function makes sure the api url (constructed using the specname) is different each time the function is called.